packages/enhanced/src/lib/container/HoistContainerReferencesPlugin.ts

```diff
@@ -4,27 +4,123 @@ import type {
   Chunk,
   WebpackPluginInstance,
 } from 'webpack';
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
 import ContainerEntryModule from './ContainerEntryModule';
 
+const runtime = require(
+  normalizeWebpackPath('webpack/lib/util/runtime'),
+) as typeof import('webpack/lib/util/runtime');
+
 /**
  * This class is used to hoist container references in the code.
  * @constructor
  */
-export class HoistContainerReferences implements WebpackPluginInstance {
+export class HoistContainerReferencesPlugin implements WebpackPluginInstance {
+  integrateChunks(
+    chunkA: Chunk,
+    chunkB: Chunk,
+    compilation: Compilation,
+  ): void {
+    const { chunkGraph, compiler } = compilation;
+    // do not sort chunk by smallest name, this will cause non-deterministic chunk integration
+
+    // if (chunkA.name && chunkB.name) {
+    //   if (
+    //     chunkGraph.getNumberOfEntryModules(chunkA) > 0 ===
+    //     chunkGraph.getNumberOfEntryModules(chunkB) > 0
+    //   ) {
+    //     // When both chunks have entry modules or none have one, use
+    //     // shortest name
+    //     if (chunkA.name.length !== chunkB.name.length) {
+    //       chunkA.name =
+    //         chunkA.name.length < chunkB.name.length ? chunkA.name : chunkB.name;
+    //     } else {
+    //       chunkA.name = chunkA.name < chunkB.name ? chunkA.name : chunkB.name;
+    //     }
+    //   } else if (chunkGraph.getNumberOfEntryModules(chunkB) > 0) {
+    //     // Pick the name of the chunk with the entry module
+    //     chunkA.name = chunkB.name;
+    //   }
+    // } else if (chunkB.name) {
+    //   chunkA.name = chunkB.name;
+    // }
+    //
+    // // Merge id name hints
+    for (const hint of chunkB.idNameHints) {
+      chunkA.idNameHints.add(hint);
+    }
+
+    // Merge runtime
+    //@ts-ignore
+    chunkA.runtime = runtime.mergeRuntime(chunkA.runtime, chunkB.runtime);
+
+    // getChunkModules is used here to create a clone, because disconnectChunkAndModule modifies
+    for (const module of chunkGraph.getChunkModules(chunkB)) {
+      // if (chunkB.name !== 'federation-runtime') {
+      chunkGraph.disconnectChunkAndModule(chunkB, module);
+      // }
+      chunkGraph.connectChunkAndModule(chunkA, module);
+    }
+
+    for (const [module, chunkGroup] of Array.from(
+      chunkGraph.getChunkEntryModulesWithChunkGroupIterable(chunkB),
+    )) {
+      // chunkGraph.disconnectChunkAndEntryModule(chunkB, module);
+      chunkGraph.connectChunkAndModule(chunkA, module);
+      // chunkGraph.connectChunkAndEntryModule(chunkA, module,chunkGroup);
```

Comment:
dont reconnect the entry moduel as a entry module, 
instead disconnect it from the chunk and reconnect it as a normal module to the runtime chunks or entrypoints 

This ensure top level exports are not compromised during re-integration

Reply:
The reason i do not disconnect the original entry module from the chunk is so that webpack still sees all modules as used - before emitting chunk assets - I delete the now integrated chunk